src/literal.ts

```diff
@@ -108,25 +137,26 @@ export default class Literal extends Node {
     }
     return new Literal(strValue, null, datatype)
   }
+
   /**
-   * @method fromValue
-   * @param value
-   * @return {Literal}
+   * Builds a literal node from an input value
+   * @param value The input value
    */
-  static fromValue (value) {
+  static fromValue(value: ValueType): Literal | Node | undefined | null {
```

Comment:
The if statements in here could also be well-served by some user-defined type guards, I think.